orttraining/orttraining/test/python/orttraining_test_orttrainer_checkpoint_functions.py

```diff
@@ -0,0 +1,351 @@
+import pytest
+from unittest.mock import patch, Mock
+from orttraining_test_orttrainer_frontend import _load_pytorch_transformer_model
+from onnxruntime.training import amp, checkpoint, optim, orttrainer
+import numpy as np
+import onnx
+import torch
+
+# Helper functions
+
+def _create_trainer(zero_enabled=False):
+    """Cerates a simple ORTTrainer for ORTTrainer functional tests"""
+
+    device = 'cuda'
+    optim_config = optim.LambConfig(lr=0.1)
```

Comment:
Different optimizers have different states. good to test them all

Reply:
These tests do not make use of the optimizers at all. All the state dictionaries are dummy dictionaries. So, the kind of optimizer does not matter in these tests.